test/server/options_impl_test.cc

```diff
@@ -324,6 +331,26 @@ TEST_F(OptionsImplTest, IncompleteComponentLogLevel) {
                           "component log level not correctly specified 'upstream'");
 }
 
+TEST_F(OptionsImplTest, InvalidLogLevel) {
+  EXPECT_THROW_WITH_REGEX(createOptionsImpl("envoy -l blah"), MalformedArgvException,
+                          "error: invalid log level specified 'blah'");
+}
+
+TEST_F(OptionsImplTest, ValidLogLevel) {
+  std::unique_ptr<OptionsImpl> options = createOptionsImpl("envoy -l critical");
+  EXPECT_EQ(spdlog::level::level_enum::critical, options->logLevel());
+}
+
+TEST_F(OptionsImplTest, WarnIsValidLogLevel) {
+  std::unique_ptr<OptionsImpl> options = createOptionsImpl("envoy -l warn");
+  EXPECT_EQ(spdlog::level::level_enum::warn, options->logLevel());
+}
+
+TEST_F(OptionsImplTest, AllowedLogLevels) {
+  EXPECT_EQ("[trace][debug][info][warning][error][critical][off][warn]",
```

Comment:
nit: from a UI perspective do you want to hack the generator function to print `"[warning|warn]"` rather than sticking `"[warn]"` at the end? I don't feel that strongly about it.

Reply:
Sure, makes sense.